tests/e2e_test.rs

```diff
@@ -185,41 +180,61 @@ async fn verify_test_contract_state(client: &ApiHttpClient) -> Result<()> {
 async fn verify_indexer(client: &ApiHttpClient) -> Result<()> {
     let response = client
         .get_indexer_contract(&ContractName("c1".to_string()))
-        .await?;
-    assert!(response.status().is_success(), "{}", response.status());
+        .await
+        .and_then(|response| response.error_for_status().context("Getting contract"));
+    assert_ok!(response);
 
-    let contract = response.json::<ContractDb>().await?;
+    let contract = response.unwrap().json::<ContractDb>().await?;
     // The indexer returns the initial state
     assert_eq!(contract.state_digest, vec![0, 1, 2, 3]);
 
     Ok(())
 }
 
+use testcontainers_modules::{
+    postgres::Postgres,
+    testcontainers::core::IntoContainerPort,
+    testcontainers::{runners::AsyncRunner, ImageExt},
+};
+
 #[tokio::test]
 async fn e2e() -> Result<()> {
     tracing_subscriber::fmt::init();
 
-    // FIXME: use tmp dir
-    let path_node1 = Path::new("tests/node1");
-    let path_node2 = Path::new("tests/node2");
+    // Start postgres DB with default settings for the indexer.
+    let _pg = Postgres::default()
+        .with_mapped_port(5432, 5432.tcp())
```

Comment:
having the test exposing 5432 port on the machine is not very handy, I have a postgres running for my manual tests, it would be better if the pg started by tests have a random port to avoid conflict with developer local setup

Reply:
yé c'est le behaviour par défaut mais faut que j'extract le port de la variable du coup, ce que j'avais eu la flemme de faire car c'était un chouilla plus difficile haha